hstream/test/HStream/RunSQLSpec.hs

```diff
@@ -130,34 +133,57 @@ handleCreateConnectorSQL sql = do
+                Just (ConstantInt s) -> Just $ fromIntegral s
+                _                    -> Nothing
+          let sk' = case typeM of
                 Just (ConstantString cType) ->
                   do
                     case cType of
                         "clickhouse" -> do
                           let username = fromMaybe "default" $ fromCOptionString (lookup "username" cOptions)
-                          let host = fromMaybe "127.0.0.1" $ fromCOptionString (lookup "host" cOptions)
-                          let port = fromMaybe "9000" $ fromCOptionString (lookup "port" cOptions)
-                          let password = fromMaybe "" $ fromCOptionString (lookup "password" cOptions)
-                          let database = fromMaybe "default" $ fromCOptionString (lookup "database" cOptions)
-                          let cli = clickHouseSinkConnector $ createClient ConnParams{
+                              host = fromMaybe "127.0.0.1" $ fromCOptionString (lookup "host" cOptions)
+                              port = fromMaybe "9000" $ fromCOptionString (lookup "port" cOptions)
+                              password = fromMaybe "" $ fromCOptionString (lookup "password" cOptions)
+                              database = fromMaybe "default" $ fromCOptionString (lookup "database" cOptions)
+                          cli <- createClient ConnParams {
                               username'     = username
                               ,host'        = host
                               ,port'        = port
                               ,password'    = password
                               ,compression' = False
                               ,database'    = database
                           }
-                          Right cli
-                        _ -> Left "unsupported sink connector"
-                _ -> Left "Invalid type in connector options"
+                          let connector = clickHouseSinkConnector cli
+                          return $ Right connector
+                        "mysql" -> do
+                          let username = fromMaybe "root" $ fromCOptionString (lookup "username" cOptions)
+                              host = fromMaybe "127.0.0.1" $ fromCOptionStringToString (lookup "host" cOptions)
+                              port = fromMaybe 3306 $ fromCOptionIntToPortNumber (lookup "port" cOptions)
+                              password = fromMaybe "password" $ fromCOptionString (lookup "password" cOptions)
+                              database = fromMaybe "mysql" $ fromCOptionString (lookup "database" cOptions)
+                          conn <- My.connect My.ConnectInfo {
+                            ciUser = username,
+                            ciPassword = password,
+                            ciPort = port,
+                            ciHost = host,
+                            ciDatabase = database,
+                            ciCharset = 33
+                          }
```

Comment:
you could refactor 
```hs
let username = fromMaybe "root" $ fromCOptionString (lookup "username" cOptions)
    host = fromMaybe "127.0.0.1" $ fromCOptionStringToString (lookup "host" cOptions)
    port = fromMaybe 3306 $ fromCOptionIntToPortNumber (lookup "port" cOptions)
    password = fromMaybe "password" $ fromCOptionString (lookup "password" cOptions)
    database = fromMaybe "mysql" $ fromCOptionString (lookup "database" cOptions)
conn <- My.connect My.ConnectInfo {
  ciUser = username,
  ciPassword = password,
  ciPort = port,
  ciHost = host,
  ciDatabase = database,
  ciCharset = 33
}
```
to
```hs
conn <- My.connect $ My.ConnectInfo <$>
  <*> (fromMaybe "root"      . fromCOptionString      $ lookup "username" cOptions)
  <*> (fromMaybe "password"  . fromCOptionString      $ lookup "password" cOptions)
  <*> (fromMaybe "127.0.0.1" . fromCOptionStringToString  $ lookup "host" cOptions)
  <*> (fromMaybe 3306        . fromCOptionIntToPortNumber $ lookup "port" cOptions)
  <*> (fromMaybe "mysql"     . fromCOptionString      $ lookup "database" cOptions)
  <*> 33
```

Reply:
Cool. Thx.

Reply:
The code is outdated, so please don't simply copy and paste, anyway, you got the idea

Reply:
I have a question. 
`
My.ConnectInfo
  (fromMaybe "root"      . fromCOptionString      $ lookup "username" cOptions)
  (fromMaybe "password"  . fromCOptionString      $ lookup "password" cOptions)
  (fromMaybe "127.0.0.1" . fromCOptionStringToString  $ lookup "host" cOptions)
  (fromMaybe 3306        . fromCOptionIntToPortNumber $ lookup "port" cOptions)
  (fromMaybe "mysql"     . fromCOptionString      $ lookup "database" cOptions)
  33
` will be fine?

If we write it in applicative style, the connect info struct will reside in a functor?